### PR TITLE
fix(Auth): Device SRP fixing comments, suggestions and invalid state transitions

### PR DESCRIPTION
**Pull request by harsh62**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/Persistence/ConfigureAuthentication.swift [hunk 1]
```diff
@@ -21,12 +21,21 @@ struct ConfigureAuthentication: Action {
...
+                    userId: authUser.userId,
+                    userName: authUser.username,
+                    signedInDate: Date(),
+                    signInMethod: .apiBased(.userSRP),
+                    cognitoUserPoolTokens: tokens)
+                authenticationEvent = AuthenticationEvent(eventType: .initializedSignedIn(signedInData))
+            }
+            catch {
+                authenticationEvent = AuthenticationEvent(
+                    eventType: .error(AuthenticationError.configuration(
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/Persistence/ConfigureAuthentication.swift:36]: not a `configuration` error, do you have something like maybe service error?

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/CredentialStorage/CredentialStoreOperationClient.swift [hunk 2]
```diff
@@ -11,9 +11,9 @@ import Security
 
 protocol CredentialStoreStateBehaviour {
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/CredentialStorage/CredentialStoreOperationClient.swift:12]: nit: Did we thought of renaming this?

**Review by royjit — APPROVED** [review]
LGTM, minor comments

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/CredentialStorage/CredentialStoreOperationClient.swift (hunk 2 above)
Comment (harsh62) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/CredentialStorage/CredentialStoreOperationClient.swift:12]: Yes.. Will do that as a separate PR of moving away from the word credential and use store..

**Pull request merged**